scipy/stats/sampling.py

```diff
@@ -49,3 +70,1262 @@
+                # to the transformed distribution
+                self._domain_pinv = d0, d1
+
+            # self._center has been set by the call self._process_config
+            # check if self._center is inside the transformed domain
+            # _domain_pinv, otherwise move it to the endpoint that is closer
+            if self._center is not None:
+                if self._center < self._domain_pinv[0]:
+                    self._center = self._domain_pinv[0]
+                elif self._center > self._domain_pinv[1]:
+                    self._center = self._domain_pinv[1]
+
+            # only suppress runtime warning in the setup if the
+            # recommended parameter ranges are used
+            if self._ignore_shape_range:
+                self.rng = NumericalInversePolynomial(
+                    dist,
+                    random_state=self.random_state,
+                    domain=self._domain_pinv,
+                    center=self._center,
+                )
+            else:
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore", category=RuntimeWarning)
+                    self.rng = NumericalInversePolynomial(
+                        dist,
+                        random_state=self.random_state,
+                        domain=self._domain_pinv,
+                        center=self._center,
+                    )
+        else:
+            raise ValueError(f"Unknown distname: {distname}")
+
+    def _set_domain_adj(self):
+        """ Adjust the domain based on loc and scale. """
+        loc = self._frozendist.kwds.get("loc", 0)
+        scale = self._frozendist.kwds.get("scale", 1)
+        lb = self._domain[0] * scale + loc
+        ub = self._domain[1] * scale + loc
+        self._domain_adj = (lb, ub)
+
+    def _process_config(self, distname, args):
+        cfg = PINV_CONFIG[distname]
+        if "check_pinv_params" in cfg:
+            if not self._ignore_shape_range:
+                if not cfg["check_pinv_params"](*args):
+                    msg = "No generator is defined for the shape parameters "
+                    msg += f"{args}. Use ignore_shape_range to proceed "
+                    msg += "with the selected values."
+                    raise ValueError(msg)
+
+        if "center" in cfg.keys():
+            if not np.isscalar(cfg["center"]):
+                self._center = cfg["center"](*args)
+            else:
+                self._center = cfg["center"]
+        else:
+            self._center = None
+        self._rvs_transform = cfg.get("rvs_transform", None)
+        self._rvs_transform_inv = cfg.get("rvs_transform_inv", None)
+        _mirror_uniform = cfg.get("mirror_uniform", None)
+        if _mirror_uniform is None:
+            self._mirror_uniform = False
+        else:
+            self._mirror_uniform = _mirror_uniform(*args)
+
+        return CustomDistPINV(cfg["pdf"], args)
+
+    def rvs(self, size=None, random_state=None):
```

Comment:
Please remove `random_state`. We have a class, hence a state.

Reply:
I agree, if needed, the attribute random_state can be changed by the user